src/Core.jl

```diff
@@ -254,24 +228,17 @@ function Optimize!(p::DSProblem)
     #TODO check that problem definition is complete
     Setup(p)
 
-    while p.status.iteration < p.iteration_limit && GetMeshSize(p) >= min_mesh_size(p)
+    while CheckStoppingConditions(p)
         OptimizeLoop(p)
     end
 
-    #TODO these will be tidied when proper stopping conditions are implemented
-    if p.status.iteration > p.iteration_limit
-        p.status.optimization_status = IterationLimit
-    end
-    if GetMeshSize(p) <= min_mesh_size(p)
-        p.status.optimization_status = PrecisionLimit
-    end
-
     Finish(p)
 end
 
 #Initialise solver
 function Setup(p)
     p.status.start_time = time()
+    SetupStoppingConditions(p)
```

Comment:
```suggestion
    _init_stoppingconditions(p)
```
I would like us to start following the JuMP style (https://jump.dev/JuMP.jl/dev/style/index.html) as much as possible when coding (since we do intend to release these packages publicly, we should have a style guid and I think the community seems to have settled around the main Julia style guide + the JuMP extensions). That means that this kind of function should be prefixed by an underscore since it is internal only and isn't one that has to be provided for each stopping condition, and also we should be avoiding the camel case.

Reply:
I've moved a few functions over to this style in f8ca62e and b3ef240.